### DISCUSSION TITLE
Problems with WEATHER_CURRENT

### DISCUSSION THREAD

**Discussion by TPTHun** [zepp-health/zeppos-forum#92] [Q&A]
Hi!
I'm trying to create a thermometer dial on a GTR3Pro watchface.
I faced several issues:

1. There is no _hmSensor_ for the actual temperature :(

2. I created an ARC_PROGRESS widget, where I was able to modify the endpoints (_start_angle, end_angle_) based on the Low and High values of _hmSensor.WEATHER_.
That's nice, but nothing similar available for current temperature.

3. I can create an IMG_POINTER or an ARC_PROGRESS widget with   _type: hmUI.data_type.WEATHER_CURRENT_   property.
But that does not work the way how I would expect.
It doesn't reflect the actual temperature value, but it somehow tells where we are between the daily temperature low and high values.
So if the daily Low is 10deg, and daily High is 20 deg, then the ARC_PROGRESS _level_ property would show:
 0% in case of 10deg
 1% for 11deg
50% for 15deg
90% for 19deg etc. 
When using WEATHER_CURRENT, WEATHER_LOW and WEATHER_HIGH types for a Widget, they all will show this very same value, instead of the actual temperature values. This is clearly a bug I beleive.

4. I cannot read out the _level_ property from an ARC_PROGRESS widget and modify an IMG_POINTER with that value.
Maybe it is possible?

Could someone please help me how to implement a simple thermometer?

**Comment by MyoungXUE** [comment]
Hello, I have already forwarded the problem to the relevant colleagues responsible for the watchface, please wait patiently.

**Comment by zhouxinhe** [comment]
Hello, 
- we can not provide weather information data to 3rd party, because of related business cooperation requirements.
- it's possible to use widgets like TEXT_IMG, TEXT_FONT to show current temperature, you can not get the number value directly.

**Reply by TPTHun** [reply]
Hi zhouxinhe!
Thank you for your answer! However I don't really understand it.
Kind of weather information do available in _hmSensor.id.WEATHER_:  the daily low and high temperatures values are available for the actual and upcoming days.
But the current temperature information is somehow special? It _can_ be written out as TEXT, but a POINTER _cannot_ show it?  :/
